mailpoet/assets/js/src/automation/integrations/mailpoet/components/form-token-field/index.tsx

```diff
@@ -0,0 +1,81 @@
+import { FormTokenField as WpFormTokenField } from '@wordpress/components';
+import { uniq } from 'lodash';
+
+export type FormTokenItem = {
+  id: number | string;
+  name: string;
+};
+
+export type FormTokenFieldProps = Omit<
+  WpFormTokenField.Props,
+  'value' | 'suggestions' | 'onChange'
+> & {
+  selected: FormTokenItem[];
+  suggestions: FormTokenItem[];
+  onChange: (values: FormTokenItem[]) => void;
+  label: string;
+  anyValue?: FormTokenItem;
+  anyValueIsDefault?: boolean;
+};
+
+export function FormTokenField({
+  label,
+  selected,
+  suggestions,
+  onChange,
+  anyValue,
+  anyValueIsDefault,
+  ...props
+}: FormTokenFieldProps): JSX.Element {
+  if (anyValue) {
+    suggestions.push(anyValue);
```

Comment:
Mutating props (well, params in general) can be dangerous. It's probably better to make it immutable (a new array). But, more importantly, this is an effect, and as such, you should never call it in the component body. It may end up adding the item many times instead of once.

Reply:
Problem outdated thanks to [710190b](https://github.com/mailpoet/mailpoet/pull/4371/commits/710190bd68357719381a9d577bc886a495f5a367)